apollo-backend/src/test/java/io/logz/apollo/DeploymentTest.java

```diff
@@ -225,4 +234,32 @@ public void testDeploymentToInactiveEnvironment() throws Exception {
         throwableAssert.isInstanceOf(Exception.class);
         throwableAssert.hasMessage("Cannot deploy. Target environment is not active.");
     }
+
+    @Test
+    public void tesIsDeploymentShouldBeCanceled() throws Exception {
+        DeploymentDao deploymentDao = standaloneApollo.getInstance(DeploymentDao.class);
+
+        TimeUnit timeUnit = standaloneApollo.getInstance(ApolloConfiguration.class).getCancelDeployment().getTimeUnit();
+        int timeout = standaloneApollo.getInstance(ApolloConfiguration.class).getCancelDeployment().getTimeout();
+
+        Deployment expiredDeployment = ModelsGenerator.createAndSubmitDeployment(apolloTestClient);
+
+        Common.waitABit((int)TimeUnit.SECONDS.convert(timeout, timeUnit)+1);
+        //Canceling an expired deployment that passed the timeout
+        expiredDeployment = apolloTestClient.cancelDeployment(expiredDeployment);
+        assertThat(expiredDeployment.getStatus()).isEqualTo(CANCELED);
+
+        Deployment deployment = ModelsGenerator.createAndSubmitDeployment(apolloTestClient);
```

Comment:
Why another?
Just create one, try to cancel, fail, wait, try to cancel, success, end test